### DISCUSSION TITLE
Disconnect from Directive

### DISCUSSION THREAD

**Discussion by JustinVoitel** [lit/lit#2674] [Ideas]
I have a use case where I want to clean up a Event Listener from the window when the element (ElementPart) is removed from the DOM in a class Directive

I would imagine there would be a method besides render and update like:

```js
render(){...}
update(){...}
disconnect(){
   window.removeEventListener("click",someEvent)
}
```

**Comment by kevinpschaaf** [comment]
For that you'll want to extend `AsyncDirective`, which adds a `disconnected` callback for this purpose: https://lit.dev/docs/templates/custom-directives/#async-directives

Note that you should also implement `reconnected` any time you implement `disconnected`, to handle the case that a directive is re-connected to the DOM after having been disconnected.

**Reply by JustinVoitel** [reply]
Didn't know about that, Thanks! 

Is there any reason why we cant disconnect with the `Directive` class ?
